sdk/tables/azure-tables/review/tables.api.review.md

```diff
@@ -0,0 +1,695 @@
+## API Report File for "@azure/tables"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+import * as coreHttp from "@azure/core-http";
+import { PagedAsyncIterableIterator } from "@azure/core-paging";
+
+// @public
+export interface AccessPolicy {
+  expiry: Date;
+  permission: string;
+  start: Date;
+}
+
+// @public
+export interface CorsRule {
+  allowedHeaders: string;
+  allowedMethods: string;
+  allowedOrigins: string;
+  exposedHeaders: string;
+  maxAgeInSeconds: number;
+}
+
+// @public
+export type CreateEntityOptions = Omit<TableInsertEntityOptionalParams, "tableEntityProperties">;
+
+// @public
+export type CreateEntityResponse = TableInsertEntityHeaders & {
+  [propertyName: string]: any;
+} & {
+  _response: coreHttp.HttpResponse & {
+    bodyAsText: string;
+    parsedBody: {
+      [propertyName: string]: any;
+    };
+    parsedHeaders: TableInsertEntityHeaders;
+  };
+};
+
+// @public
+export interface CreateTableOptions extends coreHttp.OperationOptions {
+  queryOptions?: GeneratedQueryOptions;
+  requestId?: string;
+  responsePreference?: ResponseFormat;
+}
+
+// @public
+export type CreateTableResponse = TableCreateHeaders &
+  TableResponse & {
+    _response: coreHttp.HttpResponse & {
+      bodyAsText: string;
+      parsedBody: TableResponse;
+      parsedHeaders: TableCreateHeaders;
+    };
+  };
+
+// @public
+export interface DeleteEntityOptions extends coreHttp.OperationOptions {
+  queryOptions?: GeneratedQueryOptions;
+  requestId?: string;
+  timeout?: number;
+}
+
+// @public
+export type DeleteEntityResponse = TableDeleteEntityHeaders & {
+  _response: coreHttp.HttpResponse & {
+    parsedHeaders: TableDeleteEntityHeaders;
+  };
+};
+
+// @public
+export interface DeleteTableOptions extends coreHttp.OperationOptions {
+  requestId?: string;
+}
+
+// @public
+export type DeleteTableResponse = TableDeleteHeaders & {
+  _response: coreHttp.HttpResponse & {
+    parsedHeaders: TableDeleteHeaders;
+  };
+};
+
+// @public
+export class EdmGuid {
+  constructor(guid: string);
+  get value(): string;
+  set value(guid: string);
+}
+
+// @public
+export class EdmInt64 {
```

Comment:
This is interesting! We use a `long` package in Service Bus and I seem to recall Brian pushing hard to see if we could just use number, etc...

Do you know if we have an "approved" way of representing longer number at this point?

Reply:
@richardpark-msft, your bring up a good point. We're currently having this conversation with Brian, Jeff and Matt.

It seems like we'll land on not having these classes and instead representing these "special" types with a JSON object and give the users freedom to choose the big integer library of they liking.

```typescript
{
  id: {value: "60c1ebb2-875e-4988-8610-efe28a600c08", type: "Guid"},
  viewCount: {value: "55999333", type: "Int64"},
  name: "Some entity",
  price: 123.50,
  isActive: true,
  inStockUntil: new Date("11/05/2020");
}
```

So for example if the user has BigInt they can do:

`c = obj.viewCount.type === 'Int64' ? BigInt(obj.viewCount.value) : obj.viewCount`

Note that these `Edm` objects will only be needed when users need to explicitly set `Edm.Guid` and `Edm.Int64` in the Service all the other supported types will be handled by the SDK and represented as native types.

Reply:
Are you taking community feedback on the EdmGuid and EdmInt64 class types? I've been watching this closely and really like the class representation versus the object representation. The object representation could also be used, as the classes could have the proposed properties in the object which would allow the same check of the "type" property.